packages/avatars/demo/stories/AvatarStory.tsx

```diff
@@ -9,25 +9,51 @@ import React from 'react';
 import { Story } from '@storybook/react';
 import IconUser from '@zendeskgarden/svg-icons/src/16/user-solo-stroke.svg';
 import IconSystem from '@zendeskgarden/svg-icons/src/26/zendesk.svg';
+import Clock from '@zendeskgarden/svg-icons/src/12/clock-stroke.svg';
+import ArrowLeft from '@zendeskgarden/svg-icons/src/12/arrow-left-stroke.svg';
 import { PALETTE } from '@zendeskgarden/react-theming';
 import { Avatar, IAvatarProps } from '@zendeskgarden/react-avatars';
+
 import { TYPE } from './types';
+import { BADGE_TYPE } from './data';
 
 interface IArgs extends IAvatarProps {
   type: TYPE;
+  badgeType: typeof BADGE_TYPE[number];
+  badgeChildren: IAvatarProps['badge'];
+  badgeForegroundColor: IAvatarProps['foregroundColor'];
+  badgeBackgroundColor: IAvatarProps['backgroundColor'];
+  badgeSurfaceColor: IAvatarProps['surfaceColor'];
 }
 
-export const AvatarStory: Story<IArgs> = ({ children, type, ...args }) => (
-  <Avatar
-    {...args}
-    backgroundColor={args.backgroundColor || (type === 'image' ? undefined : PALETTE.kale[800])}
-  >
-    {
-      {
-        icon: args.isSystem ? <IconSystem /> : <IconUser />,
-        image: <img alt="" src={`images/avatars/${args.isSystem ? 'system' : 'user'}.png`} />,
-        text: <Avatar.Text>{children || (args.isSystem ? 'ZD' : 'G')}</Avatar.Text>
-      }[type]
-    }
-  </Avatar>
-);
+export const AvatarStory: Story<IArgs> = ({
+  children,
+  type,
+  badge,
+  badgeType,
+  badgeChildren,
+  ...args
+}) => {
+  const child = {
+    icon: args.isSystem ? <IconSystem /> : <IconUser />,
+    image: <img alt="" src={`images/avatars/${args.isSystem ? 'system' : 'user'}.png`} />,
+    text: <Avatar.Text>{children || (args.isSystem ? 'ZD' : 'G')}</Avatar.Text>
+  }[type || 'image'];
```

Comment:
[nit] This is confusing to me. Perhaps leave the object declaration here and place the property accessing in:

```
<Avatar
    {...}
    >
      {child[type || 'image']}
    </Avatar>
```
Perhaps stick the default value of `image` in the `avatar.stories.mdx` arg declarations?

Reply:
I believe this was [how it was prior](https://github.com/zendeskgarden/react-components/blob/main/packages/avatars/demo/stories/AvatarStory.tsx#L26-L31), however I did move it around. Would it be less confusing if moved back or is it how the property is accessed that is?